judge/views/select2.py

```diff
@@ -116,6 +116,9 @@ def get_name(self, obj):
 class ContestUserSearchSelect2View(UserSearchSelect2View):
     def get_queryset(self):
         contest = get_object_or_404(Contest, key=self.kwargs['contest'])
+        if not contest.can_see_scoreboard(self.request) or contest.hide_scoreboard and contest.is_in_contest(self.request):
```

Comment:
So organizers in contests cannot search? This could be fine, but requires reflection.